### PR TITLE
Account for CHF availability

### PR DESCRIPTION
**Pull request by zyrolasting**
Re: #29

* Added a troubleshooting section
* Hosted more digests and signatures on tutorial server
* Fetched CHFs actually available in the host OpenSSL instance
* Edited the guide to say that the CHF you use may differ than the one presented in the examples

### CODE REVIEW

File: l10n/en-us.rkt [hunk 1]
```diff
@@ -487,7 +490,15 @@
        "This is likely a bug with the service.\n"
        "Please report it to the maintainer.")]
 
-  [($openssl-error args timeout exit-code output reason)
+  [($openssl:unavailable-chf requested available)
+   (let ([r-available (reverse available)])
+   (L (~a "The requested CHF " requested " is not available in your OpenSSL build.")
+      "The currently available CHFs are"
+      (~a (string-join (reverse (cdr r-available)) ",")
```
Comment (countvajhula) [inline l10n/en-us.rkt:497]:
```
$ xi do +a definition.rkt
transfer default-local.tgz: done
definition.rkt: fetched
string-join: contract violation
  expected: (listof string?)
  given: '(md4 md5 mdc2 rmd160 sha)
```
Probably needs a `symbol->string` somewhere?

**Comment by countvajhula** [comment]
After adding a strategically placed `symbol->string`, I see:
```
$ xi do +a definition.rkt
transfer default-local.tgz: done
definition.rkt: fetched
sagegerard.com:my-first-package:default:0, output default
Resolving input "default-local.tgz"

The requested CHF sha384 is not available in your OpenSSL build.
The currently available CHFs are
md4,md5,mdc2,rmd160,shaandsha1
```

I don't have sha384 in either `definition.rkt` or `xi` -- am I missing another spot?

**Comment by zyrolasting** [comment]
Check `XIDEN_TRUST_PUBLIC_KEYS`, if you've made it to that point. Beyond that, I suspect that a digest being made with `DEFAULT_CHF` popped up somewhere, but in this PR that should be bound to the one of the symbols you saw in the program output. But since you mentioned getting `null`, I'll need to rule some other potential causes out too

File: l10n/en-us.rkt (hunk 1 above)
Comment (zyrolasting) [reply l10n/en-us.rkt:497]: Done.

**Comment by countvajhula** [comment]
That fixed the earlier issue 👍 

With this code modification in place in openssl.rkt:
```
-                                    "list"
-                                    "-digest-commands"))))
+                                    "list-message-digest-commands"))))
```

I get:
```
$ xi do +a definition.rkt
transfer default-local.tgz: done
definition.rkt: fetched
sagegerard.com:my-first-package:default:0, output default
Resolving input "default-local.tgz"

default-local.tgz: fetched
integrity violation: sha1 integrity violation (bypass: -Y/--trust-any-digest/--XIDEN_TRUST_BAD_DIGEST)
```

**Comment by zyrolasting** [comment]
@countvajhula Thank you for this. I ended up pivoting to focus on #31 so it will be a bit before we sync back on this. Currently trying to get a `.dylib` and `.dll` ready. And just so you know: I might ask you to pick up from a different spot than this PR.

Thank you for your patience!

**Comment by countvajhula** [comment]
Exciting, can't wait to try it out whenever it's ready @zyrolasting . One thing though, FYI, my OS is rather old at this point and I've been getting enough reminders to upgrade that I may end up doing that relatively soon, especially since Chrome seems to have dropped support for it 😆 .

So, whenever we pick up on this, I may or may not be using the same environment as before. In either case, I'm sure it won't matter too much to the end result.

**Comment by zyrolasting** [comment]
Perfectly fine. If you don't mind, can you tell me what architecture you would be on in either case? Right now I'm targeting `x86_64`

Someone was kind enough to help add Windows support here, but we haven't tried compiling a `.dylib` on Mac OSX yet. In order for it to work on your machine, we need to be sure that we match your architecture.

**Comment by countvajhula** [comment]
Yes it's `x86_64`. Also FTR it doesn't look like Chrome dropped support after all. It's just that Chrome has been open for almost 6 months and as a result hasn't been able to auto-update in that time, and websites were complaining that my version of Chrome was unsupported! Despite these envy-inducing uptimes, I'm still planning on upgrading the OS one of these weekends.

**Comment by zyrolasting** [comment]
Closing for now due to redundancy with #34.

**Pull request closed without merging**